fog/view/server/test-utils/src/lib.rs

```diff
@@ -231,16 +231,16 @@ pub async fn assert_e_tx_out_records(
     client: &mut FogViewRouterGrpcClient,
     records: &[ETxOutRecord],
 ) -> Result<QueryResponse, Error> {
-    // Construct an array of expected results that includes both records we expect
-    // to find and records we expect not to find.
-    let mut expected_results = HashSet::new();
-    for record in records {
-        expected_results.insert(TxOutSearchResult {
+    let mut expected_results = records
+        .iter()
+        .map(|record| TxOutSearchResult {
             search_key: record.search_key.clone(),
             result_code: TxOutSearchResultCode::Found as u32,
             ciphertext: record.payload.clone(),
-        });
-    }
+            payload_length: record.payload.len() as u32,
+        })
+        .collect::<Vec<_>>();
+    expected_results.sort_by_key(|result| result.ciphertext.clone());
```

Comment:
This is a unit test so it's not super critical, but you probably don't want `sort_by_key` in this situation, since `Vec::clone()` is going to be very expensive.